### PR TITLE
test analytics: store build annotations

### PR DESCRIPTION
**Pull request by nrainer-materialize**
This requires https://github.com/MaterializeInc/materialize/pull/27546.

As a follow-up, I will try to upload annotations without markdown. This might require some refactorings.

Demo: https://buildkite.com/materialize/test/builds?branch=nrainer-materialize%3Atest-analytics%2Fstore-build-annotations-demo

### CODE REVIEW

**Review by def- — APPROVED** [review]

File: bin/ci-annotate-errors [hunk 1]
```diff
@@ -12,4 +12,9 @@
 # ci-annotate-errors - Detect errors in junit xml as well as log files during
 # CI and find associated open GitHub issues in Materialize repository.
 
-exec "$(dirname "$0")"/pyactivate -m materialize.cli.ci_annotate_errors "$@"
+HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password "$PRODUCTION_ANALYTICS_APP_PASSWORD")
```
Comment (def-) [inline bin/ci-annotate-errors:15]: I don't think the app password should be passed as a command line flag, I'd prefer to pass through PRODUCTION_ANALYTICS_APP_PASSWORD in bin/ci-builder as an allow-listed environment variable

**Review by def- — CHANGES_REQUESTED** [review]

File: bin/ci-annotate-errors (hunk 1 above)
Comment (nrainer-materialize) [reply bin/ci-annotate-errors:15]: I understand and am also not that happy about it. Though, your suggestion reduces some flexibility by kind of hardcoding the chosen password and thereby the chosen environment.
I think I will pick up your suggestion but provide the environment variable name as input argument to allow other choices as well.

Comment (nrainer-materialize) [reply bin/ci-annotate-errors]: => https://github.com/MaterializeInc/materialize/pull/27514/commits/1e622d73fb7232ad7e0fece17a497ec550973df7 and adjusted `bin/ci-annotate-errors` in an amended commit.

File: bin/ci-annotate-errors [hunk 2]
```diff
@@ -12,4 +12,9 @@
 # ci-annotate-errors - Detect errors in junit xml as well as log files during
 # CI and find associated open GitHub issues in Materialize repository.
 
-exec "$(dirname "$0")"/pyactivate -m materialize.cli.ci_annotate_errors "$@"
+HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password-env-var "PRODUCTION_ANALYTICS_APP_PASSWORD")
+
+# remove the last line, which is the mzcompose result summary
+HOSTNAME=$(echo "$HOSTNAME" | head -n 1)
```
Comment (def-) [inline bin/ci-annotate-errors:18]: This doesn't remove the last line, strictly speaking, but only keeps the first line.

Comment (nrainer-materialize) [reply bin/ci-annotate-errors:18]: True, I will adjust the comment

**Review by def- — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
This [works now!](https://buildkite.com/materialize/test/builds/83563)

**Pull request merged**